### PR TITLE
SecurityPage: functional components & custom hooks

### PR DESCRIPTION
**Pull request by thi4go**
This is another diff towards #2438 and #2439. It refactors the security page towards a more modern react approach using functional components, custom hooks and css modules. 

This diff makes the following changes:

- convert components to `.jsx`
- add `pi-ui` components where it is fit
- delete connectors
  - `signMessagePage`
  - `verifyMessagePage`

- add custom hooks
  - `useSignMessage`
  - `useValidateAddress`
  - `useVerifyMessage`

- delete css .less files
  - `SecurityCenterMessagePage.less`

UI Changes:

- receiving a signature
new
sig
old
signatureold

- verifying a valid signature
new
valid
old
validold

- verifying an invalid signature
new
invalid
old
invalidold

This is my first decrediton PR, so please let me know if this diff needs more love with styling or any code pattern that is preffered on the repo. :+1:

### CODE REVIEW

File: app/components/views/GovernancePage/Proposals/ProposalsListItem/ProposalsListItem.jsx [hunk 1]
```diff
@@ -28,6 +28,9 @@ const ProposalsListItem = ({
   const isModified =
     (!isVoting && modifiedSinceLastAccess) ||
     (isVoting && votingSinceLastAccess);
+
```
Comment (amassarwi) [inline app/components/views/GovernancePage/Proposals/ProposalsListItem/ProposalsListItem.jsx:31]: logs

Comment (thi4go) [reply app/components/views/GovernancePage/Proposals/ProposalsListItem/ProposalsListItem.jsx]: oh ty, removed them

**Comment by thi4go** [comment]
Ok mate will add that config to my ide. :+1: 

@linnutee @MariaPleshkova could you guys give a review on these UI changes? They looked a bit old fashioned for the overall design of our other apps, so I tried to improve it a bit with components we already have in pi-ui, also as part of the efforts to centralize our design needs on it. Let me know if it needs more tweaking. Thanks

**Comment by vctt94** [comment]
needs rebase

**Comment by mathcampbell** [comment]
Haven't contributed to Decred before but I'm a graphic designer and did some of the early logos used by BTC etc. 
Just signed into comment - looks good but the invalid signature should be high visibility. Something that grabs user attention, not something easily overloooked or dismissed like a notification in the corner (but equally the old design with a box underneath isn't great either. Maybe move the red line box up around the notification area. See attached image (done on my iPad in a sketch just as a indication, obviously it's not useable!)

**Comment by amassarwi** [comment]
Just as side note, not really related to this PR directly but to the security page in general and can be done in future PR of course, I noticed that
also the toggle used here which is repeating again the same styling used in other places can be a variation of pi-ui's `Toggle`.

**Comment by amassarwi** [comment]
^ fyi - @vctt94 @linnutee @thi4go

**Comment by vctt94** [comment]
Thanks for contribuiting @mathcampbell and welcome. Always good to have new contributions.

I think what you said makes sense, but I can not see the image you said.

What are your thoughts on that? @thi4go @linnutee

**Comment by mathcampbell** [comment]
Sorry GitHub was lying to me. Said it had uploaded and it hadn’t.

**Comment by mathcampbell** [comment]
Sigh. It was still lying! Sorry, Github seems a bit "off" on appending files. Sorry.
modified invalid

Anyway, I did a slightly better quality edit in Photoshop - I meant something like this. The current version has the box down below, which isn't eye-catching plus if viewed on a smaller scree might not even be visible. Your modification is good but it's too "small" - this is a very important notification, it should be highly visible. (the "success" one is fine tho!)

**Comment by thi4go** [comment]
Hey @mathcampbell thanks for chipping in. I liked the suggestions, specially highlighting the verification box area. About the invalid signature text, I'd really like to see us integrating more pi-ui components as to make design across our apps more standardized and managable. Therefore, I would suggest keeping the pi-ui badge indicator, and add the box area highlight as well (green for valid and red for invalid). What do you think @vctt94 @linnutee ?

**Comment by amassarwi** [comment]
Maybe with thinner border as well

**Comment by MariaPleshkova** [comment]
Hey @thi4go . Nice work, but got some comments though.

1. No need in these (red arrows) as we already have fields titles (green arrows)
image

2. As far as I can see you changed the size of an input text. I think it's better to leave it smaller (as it is on your screenshots called "old"). Or it's a common change for all DCR stuff?

3. Move Invalid/Valid signature signs closer to the field.
For example:
image

or
image
(the 2nd instance came to my mind when I remembered that we use similar concept for invalid data within fields, so basically if we still use it I guess we can try the same view for valid ones as well)

A screenshot of what I'm talking about:
image

**Comment by linnutee** [comment]
Yep what Maria posted above covers the states, these should be present in the master file as well. 

I would recommend not to put a huge effort in the fields just yet. 

Historically the different input and dropdown fields in decrediton seem to have the most minor issues regarding the front-end implementation. Minor details but kind of add up in the big picture. I'd think fields altogether can be slightly re-worked and simplified to fit well across pi-ui products. This is part the pi-ui worklist, which is included in a general design domain proposal coming soon.

**Comment by amassarwi** [comment]
@MariaPleshkova @linnutee thanks for your valuable input, I like the suggestion and the long term vision.
🚀 🤖

**Comment by vctt94** [comment]
Thanks, as well, I believe we now have enough to go ahead with this PR.

**Comment by thi4go** [comment]
Hey @vctt94 @MariaPleshkova updated the PR according the latest UI requests. Think this is good to go. Currently looks like this:

Screenshot from 2020-08-03 18-40-28

Screenshot from 2020-08-03 18-40-17

Again, appreciate the review on this one. :+1:

**Comment by MariaPleshkova** [comment]
@thi4go good job! 👍

File: app/style/themes/lightTheme.js [hunk 2]
```diff
@@ -31,6 +31,7 @@ const lightTheme = {
   "disabled-color": "#8997a5",
   "disabled-background-color": "#e7eaed",
   "disabled-background-color-lighter": "#edeff1",
+  "success-text-color": "#41be53",
```
Comment (vctt94) [inline app/style/themes/lightTheme.js:34]: needs to add to dark theme

**Comment by thi4go** [comment]
Thanks for testing it out @vctt94 . Comments addressed, it's fully functional now I believe :+1:

**Pull request merged**